COLLABORATOR_GUIDE.md

```diff
@@ -73,6 +73,112 @@ All pull requests that modify executable code should be subjected to
+changes to error handling may have been made. Additional CI testing may be
+required.
+
+### When breaking changes actually break things
+
+Breaking changes are difficult primarily because they change the fundamental
+assumptions a user of Node.js has when writing their code and can cause
+existing code to stop functioning as expected -- costing developers and users
+time and energy to fix.
+
+Because breaking (semver-major) changes are permitted to land in master at any
+time, it should be *understood and expected* that at least some subset of the
+user ecosystem *may* be adversely affected *in the short term* when attempting
+to build and use Node.js directly from master. This potential instability is
+precisely why Node.js offers distinct Stable and LTS release streams that carry
+varying explicit stability guarantees. Specifically, as a general rule,
+*breaking* changes should never land in a Stable or LTS branch unless such
+changes are considered to be absolutely necessary to address critical security
+issues.
+
+When any change is landed in master, and it is determined that the such
+changes *do*, in fact, break existing code, a decision may be made to revert
+those changes either temporarily or permanently. However, the decision to
+revert or not can often be based on many complex factors that are not easily
+codified.
+
+## Deprecations
+
+Deprecation refers to the identification of APIs that should no longer be used
+and that may be removed or modified in non-backwards compatible ways in a
+future major release of Node.js.
+
+Node.js uses three fundamental Deprecation levels:
+
+* Soft-Deprecation (or Docs-only deprecation) refers to elements of the Node.js
+  public API that are being staged for deprecation in a future Node.js major
+  release. An explicit notice indicating the deprecated status is added to the
+  API documentation *but no functional changes are implemented in the code*.
+  There will be no runtime deprecation warning.
+
+* Hard-Deprecation refers to the use of warnings being reported at
+  runtime when a deprecated API is used. By default, Node.js will print a
+  warning to stderr the first time a deprecated API is used. A command-line
+  switch can be used to escalate such warnings into runtime errors that will
+  cause the Node.js process to exit. As with Soft-Deprecation, the documentation
+  for the API must be updated to clearly indicate the deprecated status.
+
+* End-of-life refers to APIs that have gone through Hard-Deprecation and are
+  ready to be removed from Node.js entirely.
+
+All Deprecations *must* be handled as breaking (semver-major) changes. This
+means that while a PR adding a Deprecation can be landed in master at any time
+with appropriate review and approval, such changes are only permitted to be
+delivered in new major releases.
+
+<a id="deprecation-cycle"></a>
+A "Deprecation cycle" is one full Node.js major release during which an API
```

Comment:
If I understand correctly, EOL means it's still there, just that we aren't going to do much to support it because it is going to be removed in the next major release.

Nothing can go to EOL without going through hard deprecation first.

So, (again, if I understand correctly) the worst case scenario is:

* LTS (let's say v14 just to have numbers to work with) release has feature X
* non-LTS v15 hard deprecates the feature, so that means it prints a warning at runtime when the feature is used
* LTS v16 moves the feature to EOL. It's still there but the runtime warnings from v15 when using the feature might be more strongly worded and a date has been set for its removal from Node.js (which would presumably be the v17 release).
* non-LTS v17 is where the feature is finally removed and your code that depends on it no longer works at all.

@jasnell Am I correct above?

@jrjohnson If I _am_ correct above, does that more-or-less match what you're proposing? Or is there still a change you'd like to see?